sale_order_product_assortment/models/sale_order.py

```diff
@@ -19,61 +19,68 @@ class SaleOrder(models.Model):
 
     @api.depends("partner_id", "partner_shipping_id", "partner_invoice_id")
     def _compute_product_assortment_ids(self):
-        self.allowed_product_ids = False
-        self.has_allowed_products = False
         partner_field = (
             self.env["ir.config_parameter"]
             .sudo()
             .get_param("sale_order_product_assortment.partner_field", "partner_id")
         )
-        product_domain = []
-        if self[partner_field]:
-            filters_partner_domain = self.env["ir.filters"].search(
-                [("is_assortment", "=", True)]
-            )
-            filters_partner_domain = filters_partner_domain.filtered(
-                lambda x: x.all_partner_ids & self[partner_field]
-            )
-            if filters_partner_domain:
-                # use OR to combine allowed products
-                # and negate the blacklisted ones with AND
-                eval_product_domain = expression.OR(
-                    safe_eval(
-                        domain,
-                        {"datetime": datetime, "context_today": datetime.datetime.now},
+        filters_partner_domain = None
+        for so in self:
+            products = False
+            partner = so[partner_field]
+            if partner:
+                if filters_partner_domain is None:
+                    # search for filters only once
+                    filters_partner_domain = self.env["ir.filters"].search(
```

Comment:
Is the case where partner field is void common? Either, put the search once before the loop.

Reply:
There's at least one execution of this method (when the SO is just created from the interface, and all fields are empty) where the partner is empty